[Tan Gee Teng] iP

geetengtan opened the pull request:
### DukePro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 👍 

And it is **FREE**

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

geetengtan committed:
level-1

geetengtan committed:
level-1

geetengtan committed:
level-2

geetengtan committed:
Mark as Done

geetengtan committed:
ToDos, Events, Deadlines

geetengtan committed:
Automated Text UI Testing

geetengtan committed:
Handle Errors

geetengtan committed:
Delete

geetengtan committed:
Simplify code

geetengtan committed:
Add save feature

geetengtan committed:
Add date time formatting

geetengtan committed:
test branching

geetengtan committed:
Merge branch 'branch-Level-7'

geetengtan committed:
Merge branch 'branch-Level-8'

geetengtan committed:
Fix bug for merging

geetengtan committed:
Use More OOP

geetengtan committed:
Organize into Packages

yunbinmo commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,26 @@
+            tempStr = "X";
+        }
+
+        return "[D][" + tempStr + "] " + this.description + "(" + this.time.get().format(dateFormatter) + ")";
```
Not sure if this is a displaying problem, probably it would be better to put some line breaks to improve readability?

geetengtan replied:
yes it will!

yunbinmo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,44 @@
+package duke;
+
+import java.util.Scanner;
+
+public class Duke {
```
Probably some javadocs to describe the class would be good for future developers, this also applies to several other classes :)

geetengtan replied:
still not added yet sorry

yunbinmo commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,26 @@
+            tempStr = "X";
+        }
+
+        return "[E][" + tempStr + "] " + this.description + "(" + this.time.get().format(dateFormatter) + ")";
```
Similarly, perhaps it is better the add some line breaks to make the structure of the return message look clearer to reader.

yunbinmo commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,55 @@
+
+abstract class Task {
+
+    protected static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("LLL dd yyyy");
```
Should the constant name be all uppercase using underscore to separate words?

geetengtan replied:
you're right. will change it

yunbinmo commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+        myPrint("Bye. Hope to see you again soon!");
+    }
+
+    public static void myPrint(String toPrint) {
```
Should this method name be verbs?

yunbinmo reviewed (commented):
I like how you make most of the names sound straightforward and the code blocks are properly separated by empty lines!

geetengtan committed:
Fix bugs and Add JUnit Test

geetengtan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

geetengtan committed:
Fix Gradle bugs and add jar

geetengtan committed:
Add JavaDoc

geetengtan committed:
Fix bugs and add Find Feature

geetengtan committed:
Fix gradle properties

geetengtan committed:
Follow Coding Standard for 2103

geetengtan committed:
Merge branch 'branch-Level-9'

geetengtan committed:
Merge branch 'branch-A-JavaDoc'

geetengtan committed:
Improve code readability

geetengtan committed:
GUI

geetengtan committed:
Add Pictures

seksek13 commented on src/main/java/seedu/duke/TaskList.java:
```diff
@@ -85,4 +85,25 @@ public String count() {
 
+    public String find(String keyword) {
+        String resultStr = "";
+        boolean keywordFound = false;
```
Boolean variable should use a prefix such as is, has, was.

seksek13 commented on src/main/java/seedu/duke/Ui.java:
```diff
@@ -40,6 +44,9 @@ public void nextInput(String input, TaskList tasks) {
                 myPrint(tasks.delete(Parser.getIndex(input)));
                 break;
+            case FIND:
+                myPrint(tasks.find(Parser.getDescription(input)));
```
Maybe it will be better to change the myprint() method name to something that would explain the function better

seksek13 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,149 @@
+     * @return String that describes the number of tasks in the TaskList
+     */
+    public String count() {
+        String isSingular = "s";
```
Why does the string variable had the prefix "is", it might be confusing for others as this prefix is reserved for boolean variables. It might be better to rename this variable :)

seksek13 commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,37 @@
+        tasks.add(new Event("party ", LocalDate.of(2022, 12, 12)));
+        tasks.add(new ToDo("borrow book"));
+
+        TaskList t = new TaskList(tasks);
```
is it possible to rename this variable name so that it would be clearer in showing what you are doing?

seksek13 reviewed (commented):
Overall, methods and variables had followed the naming convention, good job!

geetengtan committed:
Improve code quality

Code was messy and might be unreadable

Code quality improvement helps for future reference and checking.

Let's
* check for error-prone practices
* delete all dead code

geetengtan committed:
Find command: make matching case insensitive

Find command is case sensitive

A case insensitive find is more user-friendly because users cannot
be expected to remember the exact case of the keywords.

Let's update the search algorithm to use case-insensitive matching

geetengtan committed:
Improve code quality

geetengtan committed:
Add assertions

geetengtan committed:
Merge branch 'branch-A-CodeQuality'

geetengtan committed:
Add Notes feature

geetengtan committed:
Add HELP feature

geetengtan committed:
Change UI and update README

geetengtan committed:
update README and CodeQuality

geetengtan committed:
fix storage bug

geetengtan committed:
Merge pull request #2 from geetengtan/branch-MergingPR

Branch merging pr

geetengtan committed:
made storage cross-platform

geetengtan committed:
fix javafx version

geetengtan committed:
fix fxml version

geetengtan committed:
update README and UG
